security/s2a/internal/crypter/counter.go

```diff
@@ -0,0 +1,41 @@
+package crypter
+
+import "errors"
+
+// counter is a 64-bit counter.
+type counter struct {
+	value         uint64
+	hasOverflowed bool
+}
+
+func newCounter() counter {
+	return counter{}
+}
+
+// val returns the current value of the counter.
+func (c *counter) val() (uint64, error) {
```

Comment:
Sorry for missing this before, please rename the APIs to `value` and `increment`. It's cleaner that way. Thanks!